cypress/tests/auth.spec.ts

```diff
@@ -166,4 +219,39 @@ describe('User Login', () => {
     cy.findByText(APP_TRANSLATIONS.support).should('exist');
     cy.findByText(APP_TRANSLATIONS.knowledgeBase).should('exist');
   });
+
+  it('should redirect authenticated user to dashboard after logging in via SSO', () => {
+    // Logs user to dashboard via API
+    loginActions.loginWithSSOAndSaveSession({ username });
+
+    cy.visit('/');
+
+    // Navigates to project listing route
+    cy.url({ timeout: 15000 }).should('include', APPLICATION_ROUTES.projects);
+
+    // Necessary API call before showing dashboard elements
+    cy.wait('@submissionList', { timeout: 15000 });
+
+    // Programmatically check for page elements based on user information.
+    cy.wait('@userInfoRoute').then(({ response }) => {
+      const orgUser = response?.body?.data?.attributes as Partial<
+        MirageFactoryDefProps['user']
+      >;
+
+      const username = orgUser?.username;
+
+      // Programmatically check for user name in navbar
+      if (username) {
+        cy.findByText(username).should('exist');
+      }
+    });
+
+    // Assertion for different dashboard elements
+    cy.findByText(APP_TRANSLATIONS.startNewScan).should('exist');
```

Comment:
These assertions are repeated from lines `215 to 220`. Putting them in the Login Actions might be better.